### DISCUSSION TITLE
Usage of the library

### DISCUSSION THREAD

**Discussion by torabian** [ncruces/go-sqlite3#15] [Q&A] [answered]
Thank you for this repository;

I wonder, if the goal of this project is to bring the sqlite full support into the webassmely?

I have done few trials, to run a golang app inside browser, but the sqlite with gorm is not really treating me happily; is there any chance we use gorm+sqlite in browser using this library at the moment or near future?

**Comment by ncruces** [comment, marked as answer]
No, the goal is not browser use.

For browser use, I suspect using `github.com/mattn/go-sqlite3` or `crawshaw.io/sqlite` with `tinygo` might be easier.

The goal is approximately the same as `modernc.org/sqlite`: SQLite bindings without `cgo`, and figure out if using Wasm is better (maintainability, safety, performance) that writing an entire C-to-Go compiler to do it.

So far: maintainability and memory safety are likely improvements; performance no, for various reasons.
